### PR TITLE
Fix issue with complex return types from UDFs

### LINKED ISSUE #1791: UDF does not support List as a return type
When I try implement a UDF that returns a list of elements I receive KsqlException: Type is not supported: java.util.List.
It is generated in [SchemaUtil.java#L446](https://github.com/confluentinc/ksql/blob/v5.0.0/ksql-common/src/main/java/io/confluent/ksql/util/SchemaUtil.java#L446)

### PR DESCRIPTION
**Pull request by rodesai**
### Description 

Use getGenericReturnType to get the return type from a loaded udf. This
ensures we get the fully parametrized type, and can use that to build the
return Schema.

Have handleParametrizedType return fully-built schemas.

fixes #1791

### Testing done 
Added unit tests

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

**Comment by rodesai** [comment]
Good point @apurvam. I'll add some if they doesn't exist already.

File: ksql-engine/src/test/resources/ToList.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package org.damian.ksql.udf;
```
Comment (big-andy-coates) [inline ksql-engine/src/test/resources/ToList.java]: package 'org.damian'????

File: ksql-engine/src/test/java/io/confluent/ksql/function/UdfLoaderTest.java [hunk 2]
```diff
@@ -93,6 +94,34 @@ public void shouldLoadFunctionsFromJarsInPluginDir() {
     assertThat(multi, not(nullValue()));
   }
 
+  @Test
+  public void shouldLoadFunctionWithListReturnType() {
+    final UdfFactory toList = metaStore.getUdfFactory("tolist");
+    assertThat(toList, not(nullValue()));
+    final KsqlFunction function
+        = toList.getFunction(Collections.singletonList(Schema.OPTIONAL_STRING_SCHEMA));
+    assertThat(
```
Comment (big-andy-coates) [inline ksql-engine/src/test/java/io/confluent/ksql/function/UdfLoaderTest.java]: nit: can we neaten up this code a little?

```
// Given:
final UdfFactory toList = metaStore.getUdfFactory("tolist");

// When:
final KsqlFunction function
        = toList.getFunction(Collections.singletonList(Schema.OPTIONAL_STRING_SCHEMA));

// Then:
assertThat(function.getReturnType(), is(
   SchemaBuilder.array(Schema.OPTIONAL_STRING_SCHEMA).build()));
```

Comment (big-andy-coates) [inline ksql-engine/src/test/java/io/confluent/ksql/function/UdfLoaderTest.java]: nit: can we neaten up this code a little?

```
// Given:
final UdfFactory toList = metaStore.getUdfFactory("tolist");

// When:
final KsqlFunction function
        = toList.getFunction(Collections.singletonList(Schema.OPTIONAL_STRING_SCHEMA));

// Then:
assertThat(function.getReturnType(), is(
   SchemaBuilder.array(Schema.OPTIONAL_STRING_SCHEMA).build()));
```

File: ksql-engine/src/test/resources/ToList.java (hunk 1 above)
Comment (big-andy-coates) [inline ksql-engine/src/test/resources/ToList.java]: package 'org.damian'???? lol

Comment (rodesai) [reply ksql-engine/src/test/resources/ToList.java]: That's what we use for the other test UDFs :)

Comment (dguy) [reply ksql-engine/src/test/resources/ToList.java]: haha - that is because i copied the code from a test project i was using...

**Comment by big-andy-coates** [comment]
@rodesai  - does this fix #1791?

**Comment by rodesai** [comment]
@big-andy-coates yeah i'll add a #fixes line

**Pull request merged**